tests/integration/test_integration.py

```diff
@@ -47,23 +48,25 @@ async def wait_for_certificate_available(
             logger.info("Action failed")
             time.sleep(1)
             continue
-        certificate = action_output.get("certificate", None)
-        if not certificate:
-            logger.info("Certificate is empty")
+        certificates = action_output.get("certificates", None)
+        if not certificates:
+            logger.info("Certificates are not available")
             time.sleep(1)
```

Comment:
are these `if not certificate`s and `time.sleep`s really necessary?

Reply:
Well yes, we want to give enough time for the certs to be provided and try until we get the certs or time out.

Reply:
I guess what I meant was does wait_for_active not also wait for `executing` to go to `idle` too?

Reply:
The charm can be in active/idle and the certs could have not yet been provided to it.